subsys/bluetooth/host/hci_core.c

```diff
@@ -7818,16 +7831,8 @@ static int le_adv_set_random_addr(struct bt_le_ext_adv *adv, uint32_t options,
 			}
 
 			*own_addr_type = id_addr->type;
-		}
 
-		if (dir_adv) {
-			if (IS_ENABLED(CONFIG_BT_SMP) &&
-			    !IS_ENABLED(CONFIG_BT_PRIVACY) &&
-			    BT_FEAT_LE_PRIVACY(bt_dev.le.features) &&
-			    (options & BT_LE_ADV_OPT_DIR_ADDR_RPA)) {
-				/* This will not use RPA for our own address
-				 * since we have set zeroed out the local IRK.
-				 */
+			if (dir_adv && options & BT_LE_ADV_OPT_DIR_ADDR_RPA) {
```

Comment:
Minor: please put `()` around bit-wise operations.

Reply:
Even when it is the only thing in the if?

Reply:
Not then, since then you'd have `(( ... ))`. Here however you're combining it with other stuff.

Reply:
Yes I am :) fixed.